### PR TITLE
[GSoC] solve method for cable class

### PR DESCRIPTION
**Pull request by Ishanned**
#### Brief description of what is fixed or changed
Implemented the solve method for the cable class. For point loads, this method computes the tensions in the segments of the cable, the reaction forces at the supports and updates the length. For distributed load, this method computes the tension at each point using tension_at() method, and the reaction forces at the supports.

#### Release Notes

* physics.continuum_mechanics
   * solve method implemented in the cable class

### CODE REVIEW

**Comment by Ishanned** [comment]
@Psycho-Pirate @AdvaitPote please have a look

File: sympy/physics/continuum_mechanics/cable.py [hunk 1]
```diff
@@ -383,3 +390,95 @@ def remove_loads(self, *args):
...
+                moment_sum_2 += self._loads['point_load'][sorted_position[i][0]][0] * cos(pi * self._loads['point_load'][sorted_position[i][0]][1] / 180) * abs(self._right_support[1] - self._loads_position[sorted_position[i][0]][1])
+                moment_sum_2 += self._loads['point_load'][sorted_position[i][0]][0] * sin(pi * self._loads['point_load'][sorted_position[i][0]][1] / 180) * abs(self._right_support[0] - self._loads_position[sorted_position[i][0]][0])
+
+            label = Symbol(sorted_position[0][0]+"_"+sorted_position[1][0])
+            y2 = self._loads_position[sorted_position[1][0]][1]
+            x2 = self._loads_position[sorted_position[1][0]][0]
+            x1 = self._left_support[0]
+            y1 = self._left_support[1]
+
+            theta = -atan((y2 - y1)/(x2 - x1))
```
Comment (AdvaitPote) [inline sympy/physics/continuum_mechanics/cable.py:471]: Try changing the name of this variable to a more descriptive one

**Comment by AdvaitPote** [comment]
Just closed and reopened to run the tests again.

**Comment by JebronLames32** [comment]
Hello @Ishanned @AdvaitPote 
I have a few doubts about what has been coded here. While creating a point load, we are mentioning the positions at which the point loads are acting.
https://github.com/sympy/sympy/blob/afc9fd2780ab9ed21cb34aa818b6a184ebc51f84/sympy/physics/continuum_mechanics/cable.py#L353-L354
However, I feel this should not be done and the force should just be given an x coordinate as the attribute with an optional y attribute. This is because the user will then be allowed to create cases which will be practally wrong. For example, the case:
image
should not be technically possible. We should return errors when such inputs are provided. I think we should allow only one of the y coordinates of the loads to be given as inputs and calculate the location of the y coordinates of the remaining loads ourselves.

Another thing I noticed is that while calculating the tension and reaction forces for the uniformly distributed load, we are taking the coordinates of the lowest point of the cable as inputs and later checking if the lowest point is consistent with the supports.
https://github.com/sympy/sympy/blob/afc9fd2780ab9ed21cb34aa818b6a184ebc51f84/sympy/physics/continuum_mechanics/cable.py#L541-L543
I feel we could be only taking the length of sag of the cable (y coordinate) as input and calculate the x coordinate ourselves with something like the code:
```python
from sympy import sqrt

lowest_y = sympify(args[0])
x1 = self._left_support[0]
y1 = self._left_support[1]
x2 = self._right_support[0]
y2 = self._right_support[1]
lowest_x = self.length / (sqrt( (y1 - lowest_y) / (y2 - lowest_y) ) + 1)
```

I have solved for the tensions and reaction forces myself using vectors (`sympy.physics.vector`)  instead of plain trigonometry. For representing the tension caused due to point loads, I have used the (`Piecewise`) function. The reason I tried  using vectors is because I felt they looked more understandable and could be more flexibly applied to calculate certain other properties of the cable in the future. I did want to clarify the above doubts before I made a pull request for you to view my code as I felt these were important points.

**Comment by AdvaitPote** [comment]
@Ishanned Let's prioritize closing/merging this PR before the GSoC coding period starts.

Also, @JebronLames32 your points are valid. @Ishanned kindly see if you can add those changes as well.

**Comment by Ishanned** [comment]
Hey @AdvaitPote, I'll soon take care of the precision errors that are causing the tests to fail.
Also, I am a bit occupied currently, so I might not be able to make the changes that @JebronLames32 has suggested. If @JebronLames32 wishes, you can make a new PR for this issue once we have sorted this one. Else, maybe @shishir-11 can take up this issue since he is going to be working on this sub-module over the summer. Let me know whoever wants to take this up, I would certainly be taking part in the discussions.

**Comment by shishir-11** [comment]
I have a few questions about calculating the y coordinates for distributed loads . Can we guess the shape of the cable or its equation. If its y coordinates form a quadratic equation we would not need to pass a lowest y coordinate on our own. 

We might be able to use or some similar formula

$$ y_{1} = ax_{1}^{2} + bx_{1}+c $$

$$ y_{2} = ax_{2}^{2} + bx_{2}+c $$

$$ length = \sqrt{1+(\frac{dy}{dx})^{2}} $$

We would be able to calculate the coordinates of the lowest point. Please correct me if I'm  wrong.

**Comment by shishir-11** [comment]
> Another thing I noticed is that while calculating the tension and reaction forces for the uniformly distributed load, we are taking the coordinates of the lowest point of the cable as inputs and later checking if the lowest point is consistent with the supports.

Since we aren't taking the length of the cable from the user for calculation , The current code works fine. I'm unable to find a method to implement what @JebronLames32 initially asked for , so i would like to request if there is any content or formula present to achieve it.

**Comment by AdvaitPote** [comment]
Cool. It has already been discussed that the priority should be to merge this PR so that @shishir-11 can get started with his project. 
Any bugs or potential changes mentioned by @JebronLames32 can be added in the future as an extended goal.

**Pull request merged**